src/LuaTransformer.ts

```diff
@@ -238,140 +247,163 @@ export class LuaTransformer {
-                statement.exportClause.elements.some(
-                    e =>
-                        (e.name !== undefined && e.name.originalKeywordKind === ts.SyntaxKind.DefaultKeyword) ||
-                        (e.propertyName !== undefined &&
-                            e.propertyName.originalKeywordKind === ts.SyntaxKind.DefaultKeyword)
-                )
-            ) {
-                throw TSTLErrors.UnsupportedDefaultExport(statement);
-            }
-
             if (!this.resolver.isValueAliasDeclaration(statement)) {
                 return undefined;
             }
 
-            const exportSpecifiers = statement.exportClause.elements.filter(e =>
-                this.resolver.isValueAliasDeclaration(e)
-            );
+            const exportSpecifiers = tsHelper.getExportable(statement.exportClause, this.resolver);
 
+            // export { ... };
             if (statement.moduleSpecifier === undefined) {
-                return exportSpecifiers.map(specifier => {
-                    let exportedIdentifier: tstl.Expression | undefined;
-                    if (specifier.propertyName !== undefined) {
-                        exportedIdentifier = this.transformIdentifier(specifier.propertyName);
-                    } else {
-                        const exportedSymbol = this.checker.getExportSpecifierLocalTargetSymbol(specifier);
-                        exportedIdentifier = this.createShorthandIdentifier(exportedSymbol, specifier.name);
-                    }
-
-                    return tstl.createAssignmentStatement(
-                        this.createExportedIdentifier(this.transformIdentifier(specifier.name)),
-                        exportedIdentifier
-                    );
-                });
+                return exportSpecifiers.map(exportSpecifier => this.transformExportSpecifier(exportSpecifier));
             }
 
-            // First transpile as import clause
-            const importClause = ts.createImportClause(
-                undefined,
-                ts.createNamedImports(exportSpecifiers.map(s => ts.createImportSpecifier(s.propertyName, s.name)))
-            );
-
-            const importDeclaration = ts.createImportDeclaration(
-                statement.decorators,
-                statement.modifiers,
-                importClause,
-                statement.moduleSpecifier
-            );
+            // export { ... } from "...";
+            return this.transformExportSpecifiersFrom(statement, statement.moduleSpecifier, exportSpecifiers);
+        } else {
+            // export * from "...";
+            return this.transformExportAllFrom(statement);
+        }
+    }
 
-            // Wrap in block to prevent imports from hoisting out of `do` statement
-            const block = ts.createBlock([importDeclaration]);
-            const result = this.transformBlock(block).statements;
+    public transformExportSpecifier(node: ts.ExportSpecifier): tstl.AssignmentStatement {
```

Comment:
All public statement visitor methods should return StatementVisitResult

Reply:
ExportSpecifier is a `Declaration`, not a `Statement`.

```ts
export { x as default, y }; // export { ExportSpecifier, ExportSpecifier };
export { x } from "default"; // export { ExportSpecifier } from "default";
```

Does the return type `tstl.AssignmentStatement` not make a lot of sense?

Reply:
It's not really a visitor in this case - currently we have only expression and statement visitors, so it should be `protected` (see `transformEntityName` for a similar case).

During #615 refactor I think we'll add a basic `transformNode` + runtime cast that would cover this better (`transformExpression` and `transformStatements` would just call `transformNode` and cast).